chcemvediet/settings/common.py

```diff
@@ -84,7 +84,7 @@
     )
 
 AUTHENTICATION_BACKENDS = (
-    u'django.contrib.auth.backends.ModelBackend',
+    u'chcemvediet.apps.accounts.backends.AdminLoginAsBackend',
```

Comment:
Ak sa admin prihlasi pomocou email + heslo, tak admin_login_as funkionalita nebude fungovat, lebo prihlasenie nastane cez `AuthenticationBackend`

Reply:
Asi sme sa nepochopili, momentalna funkcionalita:
Ak sa prihlasujem cez username+password
![Screenshot 2021-10-30 at 19-27-03 Log in Django site admin](https://user-images.githubusercontent.com/29918752/139543076-af9ec15d-68a4-4710-a042-93230dfe93e0.png)
`AdminLoginAsBackend.autheticate` je uspesne, "Admin Login As" funguje.

Pokial sa prihlasujem v admin rozhrani s email+password
![Screenshot 2021-10-30 at 19-04-27 Log in Django site admin](https://user-images.githubusercontent.com/29918752/139543108-7e754628-2b87-4a7e-994d-8092ad96326a.png)
 `AdminLoginAsBackend.autheticate` neuspeje, potom sa pouzije druhy v poradi backend `auth_backends.AuthenticationBackend` a jeho metoda `authenticate` je uspesne, ale nefunguje "Admin Login As". 

A otazka znie, ci to treba opravovat alebo to tak moze ostat.
Vyhoda tohto riesenia je, ze pri normalnom prihlasenii cez url `/en/accounts/login/` je login input `<input name="login" type="email">`. Cize normalny user musi pouzit svoj email a tym padom sa pouzije `auth_backends.AuthenticationBackend`, takze vobec nebude mat ani len pristup ku "Admin Login As" funkcionalite, bude sa pouzivat `ModelBackend.get_user` metoda a nie nami definovana.
Druha vlastnost je, ze admin musi pouzit `/admin/login` ak chce pouzit "Admin Login As". Ak by sa prihlasil cez `/en/accounts/login/` (musi cez email+password), a potom sa redirectol na `/en/admin`, nebude moct pouzit "Admin Login As".

Ak chceme aby to fungovalo aj cez email+password, mozeme: 
a) namiesto `ModelBackend` dedit `AuthetcationBackend`
```
class AdminLoginAsBackend(AuthenticationBackend):
    ...
``` 

b) implementovat vlastnu `autheticate` metodu na tento princip
```
   def authenticate(self, **credentials):
        ret = self._authenticate_by_email(**credentials)
        if not ret:
            ret = self._authenticate_by_username(**credentials)
        ...
```

Reply:
Aha. To som nevedel, ze admin sa moze prihlasit aj pomocou mailu, nie len loginu. V tom pripade by asi bolo rozumne pridat "login as" feaure pre oba backedny.

Backend `allauth.account.auth_backends.AuthenticationBackend` je tiez podedeny od `ModelBackend` ako nas `AdminLoginAsBackend`. Nestacilo by teda z nasho `AdminLoginAsBackend` spravit mixin `AdminLoginAsBackendMixin` a potom ten mixin pouzit na oba povodne backendy?

Upravene backendy potom nazvi nejak tak, by bolo jasne, ktory je ktory. Napriklad `DjangoModelBackendWithAdminLoginAs` a `AllauthAuthenticationBackendWithAdminLoginAs`.

Ak by to takto s jednym mixinom upravujucim iba `get_user` metodu nefungovalo. Tak na allauth beckend kaslime. Ale ak to fungovat bude, tak to spravme univerzalne.

Reply:
> Vyhoda tohto riesenia je, ze pri normalnom prihlasenii cez url `/en/accounts/login/` je login input `<input name="login" type="email">`. Cize normalny user musi pouzit svoj email a tym padom sa pouzije

Na toto sa nemozeme spoliehat. User si moze HTML obsah stranky upravit v browsri ako chce. Ochrana pred zneuzitim "login as" feature by mala byt v tom, ze funguje iba pre userov s nastavenym `user.is_staff`. A to by mal byt iba admin.

Reply:
:ok: 
 + Kod sa zjednodusil natolko, ze nebol problem presunut `AdminLoginAsMixin` do vseobecneho adresara `poleno`. 
Mozno by bolo vhodnejsie definovat `DjangoModelBackendWithAdminLoginAs` a `AllauthAuthenticationBackendWithAdminLoginAs` v `chcemvediet` adresari.

Reply:
@viliambalaz IMHO je lepsie ak `DjangoModelBackendWithAdminLoginAs` a `AllauthAuthenticationBackendWithAdminLoginAs` su v reusable `poleno/` adresari. Nie tu to specificke backendy pre tento projekt, su vseobecne reusovatelne. Nechaj by som ich teda tam, kde su.

Reply:
:ok: